amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -290,24 +305,16 @@ void initialize() {
                     log.info("Initializing LeaseCoordinator");
                     leaseCoordinator.initialize();
 
-                    TaskResult result;
                     if (!skipShardSyncAtWorkerInitializationIfLeasesExist || leaseRefresher.isLeaseTableEmpty()) {
                         // TODO: Resume the shard sync from failed stream in the next attempt, to avoid syncing
                         // TODO: for already synced streams
-                        for(Map.Entry<StreamIdentifier, StreamConfig> streamConfigEntry : currentStreamConfigMap.entrySet()) {
-                            final StreamIdentifier streamIdentifier = streamConfigEntry.getKey();
-                            log.info("Syncing Kinesis shard info for " + streamIdentifier);
-                            final StreamConfig streamConfig = streamConfigEntry.getValue();
-                            ShardSyncTask shardSyncTask = new ShardSyncTask(shardDetectorProvider.apply(streamIdentifier),
-                                    leaseRefresher, streamConfig.initialPositionInStreamExtended(),
-                                    cleanupLeasesUponShardCompletion, ignoreUnexpetedChildShards, 0L,
-                                    hierarchicalShardSyncer, metricsFactory);
-                            result = new MetricsCollectingTaskDecorator(shardSyncTask, metricsFactory).call();
-                            // Throwing the exception, to prevent further syncs for other stream.
-                            if (result.getException() != null) {
-                                log.error("Caught exception when sync'ing info for " + streamIdentifier, result.getException());
-                                throw result.getException();
+                        if (!waitAndCheckIfLeaseTableIsReady()) {
+                            for (Map.Entry<StreamIdentifier, StreamConfig> streamConfigEntry : currentStreamConfigMap.entrySet()) {
```

Comment:
Add a log indicating this worker is initiating the lease sync.

Reply:
Addressed